client/src/components/UI/BasePeriodSelect.vue

```diff
@@ -0,0 +1,75 @@
+<template>
+    <v-select v-model="period" :items="values" item-text="text" item-value="value" label="Time Period" />
+</template>
+
+<script>
+import { mapActions } from 'vuex';
+
+export default {
+    data() {
+        return {
+            period: 1,
+            values: [
+                { text: '1st week of January', value: 1 },
+                { text: '2nd week of January', value: 2 },
+                { text: '3rd week of January', value: 3 },
+                { text: '4th week of January', value: 4 },
+                { text: '5th week of January', value: 5 }
+            ]
+        };
+    },
+
+    watch: {
+        period() {
+            this.fetchTrafficByTime();
+        }
+    },
+
+    methods: {
+        ...mapActions({ fetchTrafficEntriesByTime: 'traffic/fetchEtriesByTime' }),
+
+        async fetchTrafficByTime() {
+            let between;
+
+            switch (this.period) {
+                case 1:
```

Comment:
const periods = [{from:xxx, to: yyy}, {from:zzz, to:ccc}...]
const between = periods[this.period - 1];

Reply:
@daniel-pupczyk-rmtcfm-com Ok.